src/classic/element/ReactElementValidator.js

```diff
@@ -194,7 +194,10 @@ function validateChildKeys(node, parentType) {
         validateChildKeys(child, parentType);
       }
     }
-  } else if (ReactElement.isValidElement(node)) {
+  } else if (
+      typeof node === 'string' || typeof node === 'number' ||
+      ReactElement.isValidElement(node)
```

Comment:
Well, they're valid _nodes_, but so are arrays and fragments. Maybe we could have isScalarNode or something but this is the logic we currently have in traverseAllChildren.